rmw_zenoh_cpp/src/detail/rmw_publisher_data.cpp

```diff
@@ -315,14 +340,52 @@ rmw_ret_t PublisherData::publish_serialized_message(
   opts.put_options.attachment = rmw_zenoh_cpp::AttachmentData(
     sequence_number_++, source_timestamp, entity_->copy_gid()).serialize_to_zbytes();
 
-  std::vector<uint8_t> raw_data(
-    serialized_message->buffer,
-    serialized_message->buffer + data_length);
-  zenoh::Bytes payload(std::move(raw_data));
+  // Get memory from SHM buffer if available.
+  if (shm && data_length >= shm->msgsize_threshold) {
+    RMW_ZENOH_LOG_DEBUG_NAMED("rmw_zenoh_cpp", "SHM is enabled.");
+
+    auto alloc_result = shm->shm_provider.alloc_gc_defrag(data_length);
+
+    if (std::holds_alternative<zenoh::ZShmMut>(alloc_result)) {
```

Comment:
`#include <variant>` for `std::holds_alternative`

Reply:
Resolved in [3f224d7](https://github.com/ros2/rmw_zenoh/pull/363/commits/3f224d749e8c58ca502d2cb60b5044654b13ec70)